payments-core/src/main/java/com/stripe/android/paymentsheet/injection/FlowControllerModule.kt

```diff
@@ -0,0 +1,224 @@
+                } ?: GooglePayRepository.Disabled
+                googlePayRepository.isReady().first()
+            },
+            workContext = Dispatchers.IO
+        )
+    }
+
+    @Provides
+    @Singleton
+    fun provideEventReporter(
+        sessionId: SessionId,
+        appContext: Context
+    ): EventReporter {
+        return DefaultEventReporter(
+            mode = EventReporter.Mode.Custom,
+            sessionId,
+            appContext
+        )
+    }
+
+    @Provides
+    @Singleton
+    fun provideSessionId() = SessionId()
+
+    /**
+     * Provides the paymentOptionActivityLauncher instance.
+     *
+     * Use [Lazy] to delay [DefaultFlowController] instance fetch to resolve its circular
+     * dependency with paymentOptionActivityLauncher.
+     */
+    @Provides
+    @Singleton
+    fun providePaymentOptionActivityLauncher(
+        activityLauncherFactory: ActivityLauncherFactory,
+        lazyDefaultFlowController: Lazy<DefaultFlowController>
+    ): ActivityResultLauncher<PaymentOptionContract.Args> =
+        activityLauncherFactory.create(
+            PaymentOptionContract()
+        ) { result ->
+            lazyDefaultFlowController.get().onPaymentOptionResult(result)
+        }
+
+    /**
+     * Provides the googlePayActivityLauncher instance.
+     *
+     * Use [Lazy] to delay [DefaultFlowController] instance fetch to resolve its circular
+     * dependency with googlePayActivityLauncher
+     */
+    @Provides
+    @Singleton
+    fun provideGooglePayActivityLauncher(
+        activityLauncherFactory: ActivityLauncherFactory,
+        lazyDefaultFlowController: Lazy<DefaultFlowController>
+    ): ActivityResultLauncher<StripeGooglePayContract.Args> =
+        activityLauncherFactory.create(
+            StripeGooglePayContract()
+        ) { result ->
+            lazyDefaultFlowController.get().onGooglePayResult(result)
+        }
+
+    @Provides
+    @Singleton
+    fun provideViewModel(viewModelStoreOwner: ViewModelStoreOwner): FlowControllerViewModel =
+        ViewModelProvider(viewModelStoreOwner)[FlowControllerViewModel::class.java]
+
+    @Provides
+    @Singleton
+    fun provideStripeApiRepository(
+        appContext: Context,
+        lazyPaymentConfiguration: Lazy<PaymentConfiguration>
+    ) = StripeApiRepository(
+        appContext,
+        { lazyPaymentConfiguration.get().publishableKey }
+    )
+
+    @Provides
+    @Singleton
+    fun providePaymentFlowResultProcessor(
+        appContext: Context,
+        viewModel: FlowControllerViewModel,
+        lazyPaymentConfiguration: Lazy<PaymentConfiguration>,
+        stripeApiRepository: StripeApiRepository
+    ): PaymentFlowResultProcessor<out StripeIntent, StripeIntentResult<StripeIntent>> {
+        return when (viewModel.initData.clientSecret) {
+            is PaymentIntentClientSecret -> PaymentIntentFlowResultProcessor(
+                appContext,
+                { lazyPaymentConfiguration.get().publishableKey },
```

Comment:
I see you're already doing what I mentioned in the other comment, so we just need to make sure the key is provided this way everywhere.